src/Common/FitOnFhir.Common/Services/AuthStateService.cs

```diff
@@ -101,38 +130,26 @@ public AuthState CreateAuthState(HttpRequest httpRequest)
 
                 if (jwtSecurityToken == default)
                 {
-                    return default;
+                    throw new AuthStateException("The security token is invalid.");
                 }
 
-                ExternalIdentifier = jwtSecurityToken.Subject;
-                ExternalSystem = jwtSecurityToken.Issuer;
-            }
-
-            var redirectUrl = HttpUtility.UrlDecode(httpRequest.Query[Constants.RedirectUrlQueryParameter]);
-
-            if (string.IsNullOrEmpty(redirectUrl))
-            {
-                string errorMessage = $"The required parameter {Constants.RedirectUrlQueryParameter} was not provided in request";
-                _logger.LogError(errorMessage);
-                throw new RedirectUrlException(errorMessage);
+                externalId = jwtSecurityToken.Subject;
+                externalSystem = jwtSecurityToken.Issuer;
             }
 
             if (!_authenticationConfiguration.ApprovedRedirectUrls.Any(url => string.Equals(url, redirectUrl, StringComparison.OrdinalIgnoreCase)))
```

Comment:
I actually think we need to check if `_authenticationConfiguration.ApprovedRedirectUrls` is null before using it, and write a test to validate the behavior. Currently, if the test is changed to return null for `ApprovedRedirectUrls`, the process crashes with a NullReferenceException.

I realize that the current implementation of the `AuthenticationConfiguration` object makes it unlikely that the value will ever be null, but depending on the implementation of another class is not the best idea.

Reply:
This should be enforced in the ARM template as a required parameter, and not in the code.

Reply:
Customer's can deploy resources without using the ARM template, they can also just modify the Function settings directly. The correct way to do this is to code defensively and just make sure the code handles the potential possibility that `_authenticationConfiguration.ApprovedRedirectUrls` can be null.